app/src/main/java/com/futurumgame/base/gameinternals/GameRoutine.java

```diff
@@ -121,6 +126,6 @@ public static void stop() {
     }
 
     public static void updateUi() {
-        currentActivity.updateUi(getCurrentWareHouse());
+        currentActivity.updateUi(current.wareHouse);
```

Comment:
This is a public static method and currentActivity does not have an initial value. As well as current. So it could be null... Null Check?

Reply:
can be ignored due to code convention this will never be null upon call, if this causes a null exception then the code somewhere else is false

Reply:
What happens with theoretical third party code that could be nullable?